api/content-script/listeners/settings-updated.ts

```diff
@@ -3,6 +3,29 @@ import { ApiUtils } from "@/api/background/utils.ts";
 
 export const watchSettingsUpdated = () => {
   storage.watch<Settings>(SETTINGS_KEY, async (newValue, oldValue) => {
+    if (oldValue && newValue) {
+      const isHostConnectedInOldValue =
+        await ApiUtils.isHostConnectedWithSettings(
+          window.location.host,
+          oldValue,
+        );
+      const isHostConnectedInNewValue =
+        await ApiUtils.isHostConnectedWithSettings(
+          window.location.host,
+          newValue,
+        );
+
+      if (isHostConnectedInOldValue !== isHostConnectedInNewValue) {
```

Comment:
If they are both `false` you want to send the message?

I think this is correct but I am a bit confused of about the business rules of this event. Which is: when should you send this event?

When the wallet's configuration switches to settings that complies with previously established connection?

Sending `kas:host_connected` does not imply that `isHostConnectedInNewValue` is true?

Anyways, if you know the logic is correct, then disregard this comment.

Reply:
> If they are both false you want to send the message?
No, it only sends the message when host connected changed. Let me show you a demo video to explain the goal I want to do in sdk:

https://github.com/user-attachments/assets/6679a677-2b52-4662-80e6-c88a71786984